phpBB/phpbb/template/context.php

```diff
@@ -264,6 +264,77 @@ public function assign_block_vars_array($blockname, array $block_vars_array)
 	}
 
 	/**
+	* Find the index for a specified key in the innermost specified block
+	*
+	* @param	string	$blockname	the blockname, for example 'loop'
+	* @param	mixed	$key		Key to search for
+	*
+	* array: KEY => VALUE [the key/value pair to search for within the loop to determine the correct position]
+	*
+	* int: Position [the position to search for]
+	*
+	* If key is false the position is set to 0
+	* If key is true the position is set to the last entry
+	*
+	* @return mixed false if not found, index position otherwise; be sure to test with ===
+	*/
+	public function find_key_index($blockname, $key = false)
+	{
+		// For nested block, $blockcount > 0, for top-level block, $blockcount == 0
+		$blocks = explode('.', $blockname);
+		$blockcount = sizeof($blocks) - 1;
+
+		$block = &$this->tpldata;
```

Comment:
You are of course right. The reason for this are two:
- I was duplicating the code in alter_block_array, https://github.com/javiexin/phpbb/blob/cff57f9076dd160b2185895f7298d4acbc5b5bb3/phpBB/phpbb/template/context.php#L367-L404, so that it is (almost) exactly the same.
- Using references does not copy the object, in this case, complex multilevel arrays, so execution time would be (much) faster, and the block is unchanged anyways...
If you still think it is better to remove the reference, I will do it from all places where it is used in this funciton (none is needed, there are 4 instances I think).